core/src/main/java/com/linecorp/armeria/server/Http1RequestDecoder.java

```diff
@@ -139,6 +166,12 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                 return;
             }
 
+            if (!webSocketContext.webSocketEstablished() &&
+                webSocketContext.lastWebSocketUpgradeRequestId() > 0) {
+                fail(id, HttpStatus.BAD_REQUEST, "Invalid decoder state", null);
```

Comment:
Could you rephrase the message if reaching here doesn't mean a bug, like `"Failed to decode a WebSocket frame"`?